dask/tests/test_distributed.py

```diff
@@ -556,3 +556,32 @@ async def test_annotation_pack_unpack(c, s, a, b):
     unpacked_hlg = HighLevelGraph.__dask_distributed_unpack__(packed_hlg)
     annotations = unpacked_hlg["annotations"]
     assert annotations == {"workers": {"n": ("alice",)}}
+
+
+@gen_cluster(client=True)
+async def test_shuffle_priority(c, s, a, b):
```

Comment:
@madsbk @mrocklin - This test seems to capture what we want: It makes sure that most of the "split" tasks are finished before any "combine" tasks start running.  This test fails for `main`, but passes with #7826 (or when "split" tasks are hacked to have a value of `0` in `dask/order.py`.  The only problem is that the test currently fails for this PR :/ - I'd be happy to investigate the cause, but if you are anxious to get something in asap, we should go with #7826 (and add a test similar to this one).

Reply:
ah - nevermind. I just forgot to stringify the keys...